newsroom/search.py

```diff
@@ -550,12 +551,30 @@ def validate_request(self, search):
                 abort(403, gettext("User does not belong to a company."))
             elif not len(search.products):
                 abort(403, gettext("Your company doesn't have any products defined."))
+            elif search.args.get("product") and not self.is_validate_product(search):
+                abort(403, gettext("Your product is not assigned to you or your company."))
             # If a product list string has been provided it is assumed to be a comma delimited string of product id's
             elif search.args.get("requested_products"):
                 # Ensure that all the provided products are permissioned for this request
                 if not all(p in [c.get("_id") for c in search.products] for p in search.args["requested_products"]):
                     abort(404, gettext("Invalid product parameter"))
 
+    def is_validate_product(self, data):
+        """
+        Check product is assigned to the company but not to the respective user.
+
+        :param SearchQuery search: The search query instance
```

Comment:
seems like a mismatch with the param name

Reply:
my bad typo